### PR TITLE
feat: Extension/admin logs (also merging of latest release/release-2.0.0)

### PR DESCRIPTION
**Pull request by Ne-wt**
discussed on disc, the ONLY thing i can think of potentially being an issue is $user not being defined in an area i may have missed though I tried my best to test everywhere I added it. May also be some spots that should have logging that I've neglected

### CODE REVIEW

**Comment by Ne-wt** [comment]
just remembered I need to add user edit logs - will add asap so this is just a draft as a ''don't forget newt''

File: app/Http/Controllers/Admin/Users/UserController.php [hunk 1]
```diff
@@ -96,6 +96,8 @@ public function postUserBasicInfo(Request $request, $name)
             flash('You cannot edit the information of a user that has a higher rank than yourself.')->error();
         }
         else {
+            if(!logAdminAction($staff, 'Editted User', 'Editted '.$user->displayname)) { flash('Failed to log admin action.')->error(); return redirect()->back();}
```
Comment (preimpression) [inline app/Http/Controllers/Admin/Users/UserController.php:99]: Typos in this file
Editted -> Edited

File: app/Services/CharacterCategoryService.php [hunk 2]
```diff
@@ -77,6 +79,8 @@ public function updateCharacterCategory($category, $data)
 
             $category->update($data);
 
+            if(!logAdminAction($user, 'Editted Character Category', 'Editted '.$category->displayName)) throw new \Exception("Failed to log admin action.");
```
Comment (preimpression) [inline app/Services/CharacterCategoryService.php:82]: Editted -> Edited

File: app/Services/CurrencyManager.php [hunk 3]
```diff
@@ -48,6 +48,7 @@ public function grantUserCurrencies($data, $staff)
 
             if($data['quantity'] < 0)
                 foreach($users as $user) {
+                    if(!logAdminAction($staff, 'Currency Debit', 'Debitted '.$data['quantity'].' '.$currency->displayName.' from '.$user->displayname)) throw new \Exception("Failed to log admin action.");
```
Comment (preimpression) [inline app/Services/CurrencyManager.php:51]: Debitted -> Debited

File: app/Services/CurrencyService.php [hunk 4]
```diff
@@ -170,23 +174,25 @@ private function populateData($data, $currency = null)
...
-            if(DB::table('shop_stock')->where('currency_id', $currency->id)->exists()) throw new \Exception("A shop currently requires this currency to purchase an item. Please change the currency before deleting it.");
+            if(DB::table('shop_stock')->where('currency_id', $currency->id)->exists()) throw new \Exception("A shop currently requires this currency to purchase an currency. Please change the currency before deleting it.");
             // Disabled for now due to issues with JSON lookup with older mysql versions/mariaDB
-            // if(DB::table('items')->where('data->resell', $currency->id)->exists()) throw new \Exception("An item currently uses this currency for its resale value. Please change the resale information before deleting this currency.");
+            // if(DB::table('currencys')->where('data->resell', $currency->id)->exists()) throw new \Exception("An currency currently uses this currency for its resale value. Please change the resale information before deleting this currency.");
```
Comment (preimpression) [inline app/Services/CurrencyService.php:187]: I don't think this was meant to be changed?

File: app/Services/GalleryManager.php [hunk 5]
```diff
@@ -169,6 +169,8 @@ public function updateSubmission($submission, $data, $user)
             // Check that there is text and/or an image, including if there is an existing image (via the existence of a hash)
             if((!isset($data['image']) && !isset($submission->hash)) && !$data['text']) throw new \Exception("Please submit either text or an image.");
 
+            if($user->isStaff) if(!logAdminAction($user, 'Editted Gallery Submission', 'Editted gallery submission '.$submission->displayName)) throw new \Exception("Failed to log admin action.");
```
Comment (preimpression) [inline app/Services/GalleryManager.php:172]: Editted -> Edited

File: app/Services/ItemService.php [hunk 6]
```diff
@@ -393,14 +406,16 @@ public function addItemTag($item, $tag)
...
-    public function editItemTag($item, $tag, $data)
+    public function editItemTag($item, $tag, $data, $user)
     {
         DB::beginTransaction();
 
         try {
             if(!$item) throw new \Exception("Invalid item selected.");
             if(!$item->tags()->where('tag', $tag)->exists()) throw new \Exception("This item does not have this tag attached to it.");
 
+            if(!logAdminAction($user, 'Editted Item Tag', 'Editted '.$tag.' tag on '.$item->displayName)) throw new \Exception("Failed to log admin action.");
```
Comment (preimpression) [inline app/Services/ItemService.php:417]: Editted -> Edited

**Review by preimpression — CHANGES_REQUESTED** [review]
Pretty much all just proofreading for typos! I checked Google on them to see if they were just Europe vs US but that didn't seem to be the case.

**Review by Draginraptor — COMMENTED** [review]
I'd been waiting for the typo fixes Uri had pointed out, but I otherwise think that the code looks fine?

**Comment by Ne-wt** [comment]
lol i forgot to fix those honestly, will get to it asap

**Review by Draginraptor — APPROVED** [review]

**Review by preimpression — APPROVED** [review]

**Pull request merged**